src/layout/Custom/useLegacyNestedTexts.ts

```diff
@@ -0,0 +1,41 @@
+import { useMemo } from 'react';
+
+import { useLanguage } from 'src/features/language/useLanguage';
+import type { FixedLanguageList } from 'src/language/languages';
+
+export function useLegacyNestedTexts() {
+  const { language } = useLanguage();
+
+  return useMemo(() => convertToLegacyNestedTexts(language), [language]);
+}
+
+interface NestedTexts {
+  [n: string]: string | NestedTexts;
+}
+
+/**
+ * Splits the text keys by '.' and creates a nested object.
+ */
+export function convertToLegacyNestedTexts(language: FixedLanguageList): NestedTexts {
```

Comment:
Isn't the `dot-object` package made for this?

Reply:
Woah, I didn't even think of that! Very good point, I'll do one more pass here.